### PR TITLE
pg config changes for UCR DBs from pgtune

### PR DESCRIPTION
**Pull request by snopoke**
* DB Version: 9.6
* OS Type: linux
* DB Type: dw
* Total Memory (RAM): 80 GB
* CPUs num: 32
* Connections num: 350
* Data Storage: ssd

max_connections = 350
shared_buffers = 20GB
effective_cache_size = 60GB
maintenance_work_mem = 2GB
checkpoint_completion_target = 0.9
wal_buffers = 16MB
default_statistics_target = 500
random_page_cost = 1.1
effective_io_concurrency = 200
work_mem = 1872kB
min_wal_size = 4GB
max_wal_size = 8GB
max_worker_processes = 32
max_parallel_workers_per_gather = 16

### CODE REVIEW

File: src/commcare_cloud/ansible/roles/postgresql_base/defaults/main.yml [hunk 1]
```diff
@@ -29,9 +29,13 @@ postgresql_maintenance_work_mem: '64MB'
 postgresql_checkpoint_completion_target: '0.5'
 postgresql_wal_buffers: '-1'
 postgresql_default_statistics_target: '100'
+postgresql_min_wal_size: ~
```
Comment (emord) [inline src/commcare_cloud/ansible/roles/postgresql_base/defaults/main.yml]: what does `~` do? I always associate it with a user's home directory

File: environments/icds-cas/inventory/postgresql.csv [hunk 2]
```diff
@@ -38,18 +38,22 @@ pgwarehouse0,100.71.184.23,postgresql,,MUMGCCWCDPRDPW00,,,,
...
-pgwarehouse0,170,0.9,1GB,16MB,500,6GB,8GB,
+group,F.var: postgresql_checkpoint_completion_target,S.var: postgresql_maintenance_work_mem,S.var: postgresql_wal_buffers,I.var: postgresql_default_statistics_target,S.var: postgresql_effective_cache_size,S.var: postgresql_max_wal_size,S.var: enable_nestloop,
+ucr_dbs,0.9,2GB,16MB,500,60GB,8GB,off,
+shard_dbs,0.9,2GB,16MB,100,36GB,,,
+main_dbs,0.7,2GB,16MB,100,36GB,,,
+synclog_dbs,0.9,2GB,16MB,100,36GB,,,
+plproxy0,0.9,2GB,16MB,100,24GB,,,
+pgwarehouse0,0.9,1GB,16MB,500,6GB,8GB,,
+,,,,,,,,
+# group vars for postgres continued,,,,,,,,
```
Comment (emord) [inline environments/icds-cas/inventory/postgresql.csv]: These changes are hard to follow. Not necessary for this PR, but maybe it would be a positive change to use comments to separate groups of variables. something like:

```
# general memory 
group, S.var: postgresql_maintenance_work_mem, ...

# checkpoints
group, F.var: postgresql_checkpoint_completion_target

# parallel
group,I.var: postgresql_max_worker_processes
```

File: src/commcare_cloud/ansible/roles/postgresql_base/defaults/main.yml (hunk 1 above)
Comment (snopoke) [reply src/commcare_cloud/ansible/roles/postgresql_base/defaults/main.yml]: This is the canonical way to represent `null` in YAML: https://yaml.org/type/null.html

**Comment by snopoke** [comment]
Re `work_mem` I'm not sure why there's a decrease but could be related to max connections. I used 350 for these calculations which is what we're seeing at the moment.

File: src/commcare_cloud/ansible/roles/postgresql_base/defaults/main.yml (hunk 1 above)
Comment (snopoke) [reply src/commcare_cloud/ansible/roles/postgresql_base/defaults/main.yml]: could change to `null` if it's clearer

Comment (emord) [reply src/commcare_cloud/ansible/roles/postgresql_base/defaults/main.yml]: if it's the canonical way I'm happy to have it. I just hadn't seen it before
